src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java

```diff
@@ -632,7 +674,52 @@ public void replaceAction(Action a) {
         }
     }
 
+    private void makeDisabled(boolean b) throws IOException {
+        if (disabled == b) {
+            return;
+	}
+        this.disabled = b;
+        if (b) {
+	    Jenkins instance = Jenkins.getInstance();
+	    if (instance != null) {
+		instance.getQueue().cancel(this);
+	    }
+	}
+        save();
+    }
+
+    /*
+     * @Overrides method added to Job<?, ?> in Jenkins core version TODO
+     */
+    public void disable() throws IOException {
+        makeDisabled(true);
+    }
+
+    /*
+     * @Overrides method added to Job<?, ?> in Jenkins core version TODO
+     */
+    public void enable() throws IOException {
+        makeDisabled(false);
+    }
+
+    @CLIMethod(name="disable-job")
```

Comment:
Have to check what actually happens here. From a quick reading of `CLIRegisterer`, looks like either this command will be broken for `WorkflowJob` or it will be broken for `AbstractProject`. Certainly deserves to be tested at least.